newlib/libc/locale/newlocale.c

```diff
@@ -113,7 +113,15 @@ newlocale (int category_mask, const char *locale,
   (void) locale;
   (void) base;
 #ifndef _MB_CAPABLE
-  return __get_C_locale ();
+  if (!strcmp (locale, "") || !strcmp (locale, "C") || !strcmp (locale, "POSIX"))
```

Comment:
These checks are already present starting at line 131. Let's just use those as they're more comprehensive and get validated when locale support is enabled.

Reply:
You would place the error return after line 141 and leave the rest of the function executed only when _MB_CAPABLE was defined.